packages/indexer-agent/src/agent.ts

```diff
@@ -202,7 +202,11 @@ class Agent {
     )
 
     const maxAllocationEpochs = timer(600_000).tryMap(
-      () => this.network.contracts.staking.maxAllocationEpochs(),
+      () =>
+        // can i pass these 2 numbers into rule creation?
+        this.indexer.maxLifetime
+          ? this.indexer.maxLifetime
+          : this.network.contracts.staking.maxAllocationEpochs(),
```

Comment:
@fordN Do you know if I could pass this value to indexing rules for when it initiate a new rule and/or in db migration? I made a placeholder of 27 but ideally it should depend on the network and the specific indexer?